lib/s3ff/view_helper.rb

```diff
@@ -32,24 +32,31 @@ def include_s3ff_templates
                 <button class="close" data-dismiss="alert" data-unique_id="{{:unique_id}}" data-target="#upload-{{:unique_id}}" style="margin-left:1em;" type="button">
                   <span aria-hidden="true">&times;</span><span class="sr-only">Remove</span>
                 </button>
-                <span class="fa fa-file-o"></span>
-                {^{:result.filename}}
+                <span class="filename">
+                  <span class="fa fa-file-o"></span>
+                  {^{:result.filename}}
+                </span>
               </div>
             </div>
           {^{else}}
-            {^{if placeholder}}
-              {^{if placeholder.url || placeholder.filename}}
-                <div class="thumbnail" id="upload-{{:unique_id}}">
-                  {^{if placeholder.url}}
-                    <img data-link="src{:placeholder.url}">
-                  {{/if}}
-                  {^{if placeholder.filename}}
-                    <div class="caption">
-                      <span class="fa fa-file-o"></span>
-                      {^{:placeholder.filename}}
-                    </div>
-                  {{/if}}
-                </div>
+            {^{if !progress_pct}}
+              {^{if placeholder}}
+                {^{if placeholder.url || placeholder.filename}}
+                  <div class="thumbnail" id="upload-{{:placeholder.id}}">
```

Comment:
For a placeholder, the `unique_id` will always be `nil`. Use `placeholder.id` instead which can be passed in from the `s3_file_field` method call.

Also it's important to be able to have the correct and unique identifier for existing files so that they can be removed correctly.

Might have also encountered a bug where the `s3_file_field` is used in a bootstrap modal and a unique id is necessary.